router/http/src/main/scala/io/buoyant/router/http/TracingFilter.scala

```diff
@@ -1,43 +1,83 @@
 package io.buoyant.router.http
 
-import com.twitter.finagle.{Service, SimpleFilter}
-import com.twitter.finagle.http.{Request, Response}
+import com.twitter.conversions.storage._
+import com.twitter.finagle.{Service, ServiceFactory, SimpleFilter, Stack, Stackable, param}
+import com.twitter.finagle.client.StackClient
+import com.twitter.finagle.http.{Request, Response, Status}
 import com.twitter.finagle.tracing.Trace
 
+object TracingFilter {
+  val role = StackClient.Role.protoTracing
+  val module: Stackable[ServiceFactory[Request, Response]] =
+    new Stack.Module1[param.Tracer, ServiceFactory[Request, Response]] {
+      val role = TracingFilter.role
+      val description = "Traces HTTP-specific request metadata"
+      def make(_tracer: param.Tracer, next: ServiceFactory[Request, Response]) = {
+        val param.Tracer(tracer) = _tracer
+        if (tracer.isNull) next
+        else (new TracingFilter).andThen(next)
+      }
+    }
+
+  val TransferEncoding = "transfer-encoding"
+}
+
 /**
  * Annotates HTTP method, uri, and status code, content-type, and content-length.
  */
-object TracingFilter extends SimpleFilter[Request, Response] {
+class TracingFilter extends SimpleFilter[Request, Response] {
+  import TracingFilter.TransferEncoding
 
   def apply(req: Request, service: Service[Request, Response]) = {
     recordRequest(req)
     service(req).onSuccess(recordResponse)
   }
 
-  private[this] def recordRequest(req: Request): Unit = {
+  private[this] def recordRequest(req: Request): Unit = if (Trace.isActivelyTracing) {
```

Comment:
Just an aesthetic comment, but it feels like the type signatures of `recordRequest` and `recordResponse` should be similar.  With your change we have:

    private[this] def recordRequest(req: Request): Unit
    private[this] val recordResponse: Response => Unit

Maybe convert `recordRequest` to a functional val too?

Also, TIOLI, but rather than checking `Trace.isActivelyTracing` in both functions, you could just do it once inside the apply method.